seismicpro/src/velocity_cube.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+import cv2
+from scipy.interpolate import interp1d, LinearNDInterpolator
+from sklearn.neighbors import NearestNeighbors
+
+from .utils import to_list
+
+
+class VelocityInterpolator:
+    def __init__(self, stacking_velocities_dict):
+        self.stacking_velocities_dict = stacking_velocities_dict
+        self.coords = np.stack(list(stacking_velocities_dict.keys()))
+        self.coords_hull = cv2.convexHull(self.coords, returnPoints=True)
+
+        self.nearest_interpolator = NearestNeighbors(n_neighbors=1)
+        self.nearest_interpolator.fit(self.coords)
+
+        # Create artificial stacking velocities in the corners of given coordinate grid in order for
+        # LinearNDInterpolator to work with a full rank matrix of coordinates
+        min_i, min_x = np.min(self.coords, axis=0) - 1
+        max_i, max_x = np.max(self.coords, axis=0) + 1
+        fake_velocities = [self._interpolate_nearest(i, x)
+                           for i, x in [(min_i, min_x), (min_i, max_x), (max_i, min_x), (max_i, max_x)]]
+        vel_data = [np.concatenate([[vel.get_coords()] * len(vel.times), vel.times, vel.velocities], axis=1)
+                    for vel in list(stacking_velocities_dict.values()) + fake_velocities]
+        vel_data = np.concatenate(vel_data)
+        self.linear_interpolator = LinearNDInterpolator(vel_data[:, :-1], vel_data[:, -1])
+
+    def is_in_hull(self, inline, crossline):
+        return cv2.pointPolygonTest(self.coords_hull, (inline, crossline), measureDist=True) >= 0
+
+    def _interpolate_linear(self, inline, crossline):
+        velocity_interpolator = lambda times: self.linear_interpolator(inline, crossline, times)
+        return StackingVelocity(interpolator=velocity_interpolator, inline=inline, crossline=crossline)
+
+    def _interpolate_nearest(self, inline, crossline):
+        index = self.nearest_interpolator.kneighbors([(inline, crossline),], return_distance=False).item()
```

Comment:
```suggestion
        index = self.nearest_interpolator.neighbors([(inline, crossline),], return_distance=False).item()
```
couldn't find documentation on this, but such spelling is suspicious. Did you run this piece of code?

Reply:
I did, it works :cry: